src/core/include/openvino/core/node.hpp

```diff
@@ -207,6 +207,7 @@ class OPENVINO_API Node : public std::enable_shared_from_this<Node> {
     virtual bool evaluate_upper(ov::TensorVector& output_values) const;
     virtual bool evaluate_symbol(TensorSymbolVector& output_symbols) const;
 
+    virtual bool can_constant_fold(const OutputVector& inputs_values);
```

Comment:
```suggestion
    virtual bool can_constant_fold(const OutputVector& inputs_values) const;
```
The validation should not change the Node state